### DISCUSSION TITLE
Rendering of Shapenet models looks terrible

### DISCUSSION THREAD

**Discussion by MathieuLeclercq** [facebookresearch/pytorch3d#1107] [Q&A] [answered]
Hello,
I've been using Pytorch3d renderers for a 3D segmentation task and it worked perfectly. I am now trying to set up a classification task using the same code for the rendering, and the  [Shapenet database](https://shapenet.org/) as input data (.vtk files). The renders look wrong and I cannot tell what is happening. The shapes are correct but the normals look very wrong.
![sofa](https://user-images.githubusercontent.com/56450839/157283841-e740aef6-d20d-4dec-9fff-b845ce354c8a.png)
![chair](https://user-images.githubusercontent.com/56450839/157283843-df1dc86f-efde-4890-a49c-38d6be1a9bae.png)

I tried changing the lights (from AmbientLights to PointLights) and the shaders (SoftPhong and HardPhong) but the results are the same.

The notebook that I use for running the tests can be found [here.](https://github.com/MathieuLeclercq/fly-by-cnn/blob/master/src/py/notebooks/shapenet_train_pytorch3d_tests.ipynb)

Have you ever experienced this kind of behavior?

**Comment by gkioxari** [comment, marked as answer]
ShapeNet's CAD models are quite imperfect and this is usually demonstrated when visualizing normals or rendering textures (which depend on normals). This is why many researchers remesh the ShapeNet CAD models, either by smoothing them or with other techniques. Look around and you will find them.
